src/utils/getTreeNames.ts

```diff
@@ -0,0 +1,43 @@
+import type {TreeNode} from "@/features/DecisionTreeGenerator/type/TreeNode";
+
+export const getTreeNames = (tree: { children: TreeNode['children'], name?: TreeNode['name'], attributes?: TreeNode['attributes'] }, attributeAccumulator: string[] = []): string[] => {
+    // if juste first Element
+    if (!tree.children.length && !attributeAccumulator.length && tree?.name) {
+        return [tree.name];
+    }
+
+    // if last element
+    if (!tree.children.length) {
+        return attributeAccumulator;
+    }
+
+    // add first tree name
+    if (tree.name) {
+        attributeAccumulator = [...attributeAccumulator, tree.name];
+    }
+
+
+    const result = tree.children.map(({name, children}) => {
+        // if More than on children (decision section)
+        if (children.length > 1) {
+            const results = children.map(val => getTreeNames({children: val.children}))
+            const flattenResult = results.flat();
+
+            return {names: [...attributeAccumulator, ...flattenResult, name], rest: {children: []}}
+        }
+
+        return {names: [...attributeAccumulator, name], rest: {children}}
+
+    })[0];
+
+    return getTreeNames(result.rest, result.names)
+}
+
+export const hasUniqueName = (array: string[]): boolean => {
+    const uniqueName = new Set(array)
```

Comment:
Ça c'est vraiment nice pour checker un element unique dans un array, il y a pas mieux 💪
Par contre c'est mieux de faire un shallow copy de ton array :
```suggestion
    const uniqueName = new Set([...array]);
```